File: algorithms/19_lua/math/catalan.lua
-- Catalan numbers by the recurrence.
local c = {}
for i = 1, 11 do c[i] = 0 end
c[1] = 1
for n = 1, 10 do
  local s = 0
  for k = 0, n - 1 do
    s = s + c[k + 1] * c[n - k]
  end
  c[n + 1] = s
end
assert(c[6] == 42 and c[11] == 16796, "wrong Catalan numbers")
print("catalan(10)=" .. c[11])

File: algorithms/02_c/math/catalan.c
/* Catalan numbers by the recurrence. */
#include <stdio.h>
int main(void) {
    long c[11] = {0};
    c[0] = 1;
    for (int i = 1; i <= 10; i++) {
        long s = 0;
        for (int j = 0; j < i; j++) s += c[j] * c[i - 1 - j];
        c[i] = s;
    }
    if (c[5] != 42 || c[10] != 16796) return 1;
    printf("catalan(10)=%ld\n", c[10]);
    return 0;
}
